Fix setIdentity

jaj22 opened the pull request:
### What type of PR is this.
1. [X] Bug
2. [ ] Change
3. [ ] Enhancement

### What have you changed and why?
We used BIS_fnc_setIdentity to set voices and faces globally, but it didn't work correctly because it assumes that the initial call returns false for `isRemoteExecuted`. Because the remote execution flag is passed through spawns (wtf BIS), most Antistasi code actually returns true.

Rewrote with an explicit global/local split. Also skips the reflection-to-server, as that wasn't necessary. You can run JIP remoteExecs from anywhere.

Also changed the name code to use the proper function rather than the one that only sets first/last names, so we no longer need the hack for Petros and future unit naming code will be easier.

One somewhat unwanted effect inherited from the original code is that it continually grows the JIP queue until someone new connects and clears out the dead JIPs. A lot of work to fix though.

Cases tested:
1. Units created on client sync correctly to server. Should also apply to HCs as all isRemoteExecuted checks were removed.
2. Units created on server sync correctly to client.
3. Units sync correctly on JIP, regardless of creation location.
4. JIP clearance for deleted units works.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.
1. [ ] Have you loaded the mission in LAN host?
2. [X] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

killerswin2 commented on A3A/addons/core/functions/Utility/fn_setIdentity.sqf:
```diff
@@ -0,0 +1,20 @@
+
+Arguments:
+    <OBJECT> Object to set identity for
+    <STRING> Optional: Face of unit
```
kill the other params

killerswin2 replied:
disregard this

killerswin2 commented on A3A/addons/core/functions/Utility/fn_setIdentity.sqf:
```diff
@@ -0,0 +1,20 @@
+/*
+    Set face, voice, pitch and name of unit. Global effect and JIP-safe.
+
```
`Wrapper for remoteExec of A3A_fnc_setIdentityLocal`

jaj22 replied:
No, same reason as the disregard of the other suggestion. You don't force people to go read the spec of internal functions.

killerswin2 commented on A3A/addons/core/functions/Utility/fn_setIdentity.sqf:
```diff
@@ -0,0 +1,20 @@
+
+if (isNull _unit) exitWith {};
+private _JIPID = "identity_" + netId _unit;
+[_JIPID] + _this remoteExec ["A3A_fnc_setIdentityLocal", 0, _JIPID];
```
```suggestion
([_JIPID] + _this) remoteExec ["A3A_fnc_setIdentityLocal", 0, _JIPID];
```

killerswin2 commented on A3A/addons/core/functions/Utility/fn_setIdentityLocal.sqf:
```diff
@@ -0,0 +1,25 @@
+/*
+    Set face, voice, pitch and name of unit locally, and clear JIP if no longer present
```
```suggestion
    Sets face, voice, pitch and name of unit locally if they are passed in, and clear JIP if no longer present
```

killerswin2 commented on A3A/addons/core/functions/Utility/fn_setIdentityLocal.sqf:
```diff
@@ -0,0 +1,25 @@
+/*
+    Set face, voice, pitch and name of unit locally, and clear JIP if no longer present
```
```suggestion
    Sets face, voice, pitch and name of unit locally if they are passed in, and clear JIP if no longer present
```

jaj22 replied:
Surprised you didn't pull up the least accurate word, which is "and" :P

killerswin2 reviewed (changes requested):
Nice work. Just a few changes

Lazejun reviewed (approved):
Besides Killers mimimi about Comments/headers tested it with TADST, works.

jaj22 commented:
- Added a Deleted EH to prevent JIP queue bloating and subsequent remoteExec spam.
- Added brackets to make precedence clearer.
- Improved description on setIdentityLocal.
